arbos/util/tracing.go

```diff
@@ -53,8 +53,20 @@ func NewTracingInfo(evm *vm.EVM, from, to common.Address, scenario TracingScenar
 	}
 }
 
-func (info *TracingInfo) RecordStorageGet(key common.Hash) {
+func (info *TracingInfo) RecordStorageGet(key, mappedKey common.Hash) {
 	tracer := info.Tracer
+	// RecordStorageGet is only called from arbos storage object, meaning the SLOAD opcode tracing in the scenario TracingDuringEVM
+	// has no impact at all as the caller address (scope.Contract) and the key being passed dont associate with each other, instead
+	// the key corresponds to types.ArbosStateAddress. For this exact reason, when RecordStorageGet is called from inside arbos storage,
+	// other tracers should implement their own CaptureArbitrumStorageGet functions irrespective of tracing scenario to remove dependency
+	// on opcode tracing.
+	// ...
+	// Since CaptureArbitrumStorageGet is only implemented for prestateTracer there's no harm in calling it along with opcode tracing
+	// during TracingDuringEVM scenario (as prestate tracer fails to record any meaningful change due to the reason given above),
+	// prestateTracer now supports in removing opcode tracing from this function but other tracers don't, and that should be changed
+	if tracer.CaptureArbitrumStorageGet != nil {
```

Comment:
When we're DuringEVMStorage the right thing already happends in the prestate (and others) thanks to us simulating the SLOAD instruction.
We only need a solution for before/after stages because simulating a SSTORE/SLOAD while not in evm execution might break some stuff.

Reply:
Fixed my implementation